ImmuDB4Net/ImmuClient.cs

```diff
@@ -140,42 +176,47 @@ private void StopHeartbeat()
 
     public async Task Open(string username, string password, string defaultdb)
     {
-        if (session != null)
+        if (ActiveSession != null)
         {
             throw new InvalidOperationException("please close the existing session before opening a new one");
         }
-        Connection = ConnectionPool.Acquire(this);
-        session = await SessionManager.OpenSession(Connection, username, password, defaultdb);
+        Connection = ConnectionPool.Acquire(new ConnectionParameters
+        {
+            Address = GrpcAddress,
+            ShutdownTimeoutInSec = ConnectionShutdownTimeoutInSec
+        });
+        ActiveSession = await SessionManager.OpenSession(Connection, username, password, defaultdb);
```

Comment:
Between this line and the check at the beginning of the `Open` task there could be another thread that did change the ActiveSession (like two threads at the same time closing and Opening sessions). We should add additional check here if the `ActiveSession` is still null here to prevent race conditions. We could also set some placeholder at the beginning to *reserve* write access to the `ActiveSession` object.

Reply:
I've added a better synchronized support for Open, I might be needed to add IDisposable pattern to ImmuClient, though.